proposals/0001-gep.md

```diff
@@ -0,0 +1,198 @@
+
+As any other config, users should be able to easily apply new changes to the config. 
+The API keys may need to be rotated from time to time.
+
+Finally, the more configs we expose, the harder may become to understand the overall config
+and find the right setting to adjust. So the config structure is an important thing here to eliminate this fatigue.
+
+### Requirements
+
+- R1: Config format must be readable
+- R2: Config format must allow comments
+- R3: Config format must be `git diff`-able and versioned
+- R4: Config must be easy to understand and reason about
+- R5: Config must allow to include sensitive values from environment variables
+- R6: Config may allow to include sensitive values from other plain files
+- R7: Config should be easy to change and operate on
+
+## Design
+
+There are basically a few sources of configurations possible:
+- via Glide CLI params & args
+- via environment variables
+- via a separate config file
+- via a combination of CLI, environment variables, and/or config file
+
+Setting complex nested configs via CLI or environment variables is usually a tedious error-prone process.
+So Glide will try to leverage the file as config approach as long as possible.
+
+### The Config Source
+
+Glide config is represented as a YAML file (R1, R2, R3). 
+YAML is a human-friendly version of JSON that supports comments and some useful magic like (including environment variables).
+Go has good support for the YAML format.
+
+### The Config Structure
+
+The config structure uses Glide's internal logical structure that we are going to communicate in Glide docs (R4).
+So learning more about how Glide works should give enough context to understand the config. 
+The opposite should be true as well, so knowing how the config looks will give some understanding about Glide's overall design.
+This is a route that the OTEL Collector config takes.
+
+Glide should allow you to pull values of configs from environment variables (see the config example below).
+This helps to keep sensitive information (R5) out of the config file that avoids leakages.
+
+### Config Ops
+
+Glide should be able to automatically detect changes to the config (especially, our domain-related configs like routing) and
+reload its internal representation together with the related state (R7). 
+In Kubernetes, configs are going to be represented as a configmap mounted into Glide pods. 
+On configmap change, the mounted file changes as well. So we need to watch for the file changes there. 
+Outside of Kubernetes, we may support config reloading on SIGHUP signals.
+
+### Config Example
+
+```yaml
+telemetry:
+  log_level: info
+  log_format: plain # plain, JSON
+  # other configs
+
+api:
+  http:
+    listen_addr: 0.0.0.0:7685
+    max_body_size: "2Mi"
+    tls:
+      ca_path:
+      cert_path:
+    # other configs
+
+  grpc:
+    listen_addr: 0.0.0.0:7686
+    tls:
+      ca_path:
+      cert_path:
+    # other configs
+
+gateway:
```

Comment:
Here is a schema I am thinking:

(trying out composites instead of pools...let me know what you think)

```yaml
gateway:
  composites:
    - compName: my-app-one
      balancing: priority # round-robin, weighted-round-robin, priority, least-latency, priority, etc.
      models:
        - provider: openai # anthropic, azureopenai, gemini, other providers we support
          model: gpt-3.5-turbo
          api_key: ${env:OPENAI_API_KEY}
          default_params:
            temperature: 0
        - provider: anthropic
          model: claude-2
          apiKey: ${env:ANTHROPIC_API_KEY}
          default_params: # set the default request params
            temperature: 0
    - compName: my-app-two
      balancing: least-latency # round-robin, weighted-round-robin, priority, least-latency, priority, etc.
      models:
        - provider: openai # anthropic, azureopenai, gemini, other providers we support
           model: gpt-4-1106-preview
           api_key: ${env:OPENAI_API_KEY}
           default_params:
             temperature: 0
        - provider: openai
           model: gpt-3.5-turbo
           default_params:
             temperature: 0.1

```

Reply:
Is there a case where an app might use multiple composites/pools?